### PR TITLE
Compile using ASAN on the github CI

### PR DESCRIPTION
**Pull request by powerboat9**
This adds a job for running GCC with ASAN to the github CI

### CODE REVIEW

**Review by CohenArthur — APPROVED** [review]
great stuff! thanks a lot!

**Comment by powerboat9** [comment]
Not working yet, having trouble with linker during link tests

**Comment by powerboat9** [comment]
Alright, looks like things are working now

**Comment by powerboat9** [comment]
It looks like the execute tests are failing because runtime libraries compiled with ASAN require that the programs using them are compiled with ASAN.

File: .github/configure-san [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+
+# This should be run by github CI
+# However, running it locally (if it works on your PC) might be useful
+
+export GCC_LIBS=$(gcc -print-search-dirs | head -n1 | sed 's/install: //')
+export COMMON="-fsanitize=address"
+
+CONFIGURE=$(dirname $(dirname $0))/configure
+
+$CONFIGURE CFLAGS="$COMMON -B$GCC_LIBS" \
```
Comment (thesamesam) [inline .github/configure-san:15]: Consider using `--with-build-config="bootstrap-asan"` (or whatever it is) instead, but not sure how to handle tests (some DejaGnu hackery may be needed there).

Comment (powerboat9) [reply .github/configure-san:15]: `bootstrap-asan` doesn't work with `--disable-bootstrap`, as far as I can tell

**Comment by powerboat9** [comment]
Should be ready for review again

File: .github/configure-san (hunk 1 above)
Comment (thesamesam) [reply .github/configure-san]: Ah, you're right, but I see you've figured it out ;)

**Review by thesamesam — APPROVED** [review]
LGTM. Thanks!

**Comment by CohenArthur** [comment]
Ah, I think the test failures might be due to the execute testsuite running but not linking with ASAN? this is something that happens to me when I do local ASAN builds. If that is the case, then a solution would be to either run only the compile.exp testsuite with `RUNTESTFLAGS` or figure out how to run the `execute.exp` testsuite in compile-only mode, without linking/running the final binary. but I don't know if dejagnu supports this

**Comment by powerboat9** [comment]
I managed to get the execute tests working, and it seems like the remaining test failures are down to bugs in GCCRS

**Comment by powerboat9** [comment]
The ASAN check *should* pass now

**Review by P-E-P — APPROVED** [review]

**Pull request merged**